cloud/endagaweb/tasks.py

```diff
@@ -439,3 +439,124 @@ def req_bts_log(self, obj, retry_delay=60*10, max_retries=432):
         raise
     finally:
       obj.save()
+
+
+@app.task(bind=True)
+def subscriber_validity_state(self):
+    """ Updates the subscribers state to inactive/active/"""
+
+    today = django.utils.timezone.now()
+    subscribers = Subscriber.objects.filter(
+        number__valid_through__lte=today)
+    today = today.date()
+    for subscriber in subscribers:
+        try:
```

Comment:
you should have a loop on each of the subscriber's number. there can be more than the one at index 0